docs/spec/auth/oauth.md

```diff
@@ -0,0 +1,122 @@
+<!--[metadata]>
++++
+title = "Oauth2 Token Authentication"
+description = "Specifies the Docker Registry v2 authentication"
+keywords = ["registry, on-prem, images, tags, repository, distribution, oauth2, advanced"]
+[menu.main]
+parent="smn_registry_ref"
++++
+<![end-metadata]-->
+
+# Docker Registry v2 authentication using OAuth2
+
+This document describes support for the OAuth2 protocol within the authorization
+server. [RFC6749](https://tools.ietf.org/html/rfc6749) should be used as a
+reference for the protocol and HTTP endpoints described here.
+
+## Refresh token format
+
+The format of the refresh token is completely opaque to the client and should be
+determined by the authorization server. The authorization should ensure the
+token is sufficiently long and is responsible for storing any information about
+long-lived tokens which may be needed for revoking. Any information stored
+inside the token will not be extracted and presented by clients.
+
+## Getting a token
+
+POST /token
+
+#### Headers
+Authorization headers
+Content-Type: application/x-www-form-urlencoded
+
+#### Post parameters
+
+<dl>
+    <dt>
+        <code>grant_type</code>
+    </dt>
+    <dd>
+        (REQUIRED) Type of grant used to get token. When getting a refresh token
+        using credentials this type should be set to "password" and have the
+        accompanying basic auth header. Type "authorization_code" is reserved
+        for future use for authenticating to an authorization server without
+        having to send credentials directly from the client. When requesting an
+        access token with a refresh token this should be set to "refresh_token".
+    </dd>
+    <dt>
+        <code>service</code>
+    </dt>
+    <dd>
+        (REQUIRED) The name of the service which hosts the resource to get
+        access for. Refresh tokens will only be good for getting tokens for
+        this service.
+    </dd>
+    <dt>
+        <code>client</code>
```

Comment:
Discussed further and decided it would be OK to update this to use the `client_id` field as described in the spec. The default `client_id` value used by engine may be added here as well. Further supporting `client_id` with `client_secret` could be be left open for a possible future change, which would also require docker engines being able to configure those values for the auth domain.